messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java

```diff
@@ -161,8 +162,41 @@ public ActiveMQServerService(Configuration configuration, PathConfig pathConfig)
         return dataSource;
     }
 
-    Map<String, JChannel> getChannels() {
-        return channels;
+    JChannel getChannel(String name) throws Exception {
+        if (channels.containsKey(name)) {
+            return channels.get(name);
+        }
+        if (jgroupsChannels.containsValue(name)) {
+            Optional<String> groupName = jgroupsChannels.entrySet().stream().filter(entry -> name.equals(entry.getValue())).map(entry -> entry.getKey()).findFirst();
```

Comment:
hum the map is only written when the server is added so it is not modified. For the channels one I think you are right thought